src/vm/peimage.cpp

```diff
@@ -909,73 +909,149 @@ PTR_PEImageLayout PEImage::GetLayoutInternal(DWORD imageLayoutMask,DWORD flags)
     {
         _ASSERTE(HasID());
 
-        if (imageLayoutMask&PEImageLayout::LAYOUT_MAPPED)
+        BOOL bIsMappedLayoutSuitable;
+        BOOL bIsFlatLayoutSuitable;
+
+#ifdef PLATFORM_UNIX
+        bIsMappedLayoutSuitable = ((imageLayoutMask & PEImageLayout::LAYOUT_MAPPED) != 0);
```

Comment:
I think you can move both the statements out of the define and change the define to be:

```
#if !defined(PLATFORM_UNIX)
if (bIsMappedLayoutSuitable)
{
    bIsFlatLayoutSuitable = FALSE;
}
#endif
```